TODO.md

```diff
@@ -0,0 +1,69 @@
+# TODO
+remaining tasks for the `0.2` release, in order of priority
+
+## major
+* [ ] move behaviours (getting, updating, etc.) to library methods
+* [ ] validation (schema, column, relationship)
+* [ ] declarative entity definitions
+* [ ] schema validation
+* [ ] documentation
+* [ ] callbacks
+* [ ] pagination
+* [ ] hateoas-style support
+
+### move behaviours
+at the moment, when you generate your code, a lot of functionality
+gets put in the restplus endpoint. this results in huge diffs between
+versions and is tiresome to test.
+
+it is proposed that this code moves into a library function which then
+gets imported by the endpoint. this will make the code easier to test,
+and will result in a smaller generated codebase.
+
+### validation
+there's loads that could be done to validate a schema. the most interesting
+part is probably the relationships - to do this we would need a two-pass
+implementation where the first pass builds up an idea of what entities are
+present and how they relate to one another, and the second pass
+verifies that these relationships are possible.
+
+error reporting should be as approachable and helpful as possible. see
+[elm](https://elm-lang.org/) for inspiration.
+
+### declarative entity definitions
```

Comment:
I think this is hard to get right. There is a lot of additional information involved in putting together a a set of entities and decorator based interfaces are notoriously hard to get right, particularly if they're dealing with a lot of data.

Reply:
think this would be a much nicer interface than the current declarative approach. with the decorators I only mean for collecting the classes - the alternative to this is doing it in the SQLAlchemy style which I think is a lot less pleasant. would be fun to spike this out to see how it looks

Reply:
maybe I need to see it. It would be nice if it could be implemented initially as an additional interface that produces the current interface. That way it could introduced in a backwards compatible way.